### ISSUE TITLE
(v2) Koa hangs on errors

### ISSUE THREAD

**Issue by tnunamak** [pinojs/koa-pino-logger#4]
Using this example from the README (modified with the `new` keyword for Koa 2), my server logs an error when I hit http://localhost:3000, but it does not respond and the request times out.

```javascript
'use strict'

var Koa = require('koa')
var logger = require('koa-pino-logger')

var app = new Koa()
app.silent = true //disable console.errors
app.use(logger())

app.use(function * () {
  this.body = 'hello world'
  throw Error('bang!')
})

app.listen(3000)
```

**Comment by davidmarkclements** [comment]
Hi @tnunamak thanks for reporting. Just to be sure, this behavior is happening with the module installed like so:

```sh
npm install koa-pino-logger@2
```

So you're using the v2 logger for v2 koa?

Secondly, can you confirm that the behavior is different if you remove the logger middleware?

**Comment by tnunamak** [comment]
Yep, I just confirmed the following from my shell history. And the server definitely responds with an internal error successfully, without the middleware.

```bash
npm install koa@next
npm install koa-pino-logger@2 --save
```

**Comment by eudaimos** [comment]
@tnunamak if you are using Koa v2 then why is your handler a generator function?  Is that because the example code from `koa-pino-logger` is written that way?

I would expect (and I write my) Koa v2 handlerse using `async` functions and the body is attached to `ctx.res` or `ctx` (shorthand), i.e. the handler should be written as:
```javascript
app.use(async function (ctx, next) {
  ctx.body = 'hello world';
  throw Error('bang!');
});
```

**Comment by davidmarkclements** [comment]
@eudaimos are you saying it doesn't hang for you when written with async func? 

in that case, will happily take PR for readme

**Comment by laggingreflex** [comment]
I experienced this too. Should be fixed by #7

**Comment by davidmarkclements** [comment]
fixed in #7
